ENIGMAsystem/SHELL/Widget_Systems/xlib/dialogs.cpp

```diff
@@ -146,14 +161,86 @@ static pid_t process_execute(const char *command, int *infp, int *outfp) {
   return pid;
 }
 
-// set dialog transient.
-static void force_window_of_pid_to_be_transient(Display *display, pid_t pid) {
+#if CURRENT_PLATFORM_ID == OS_MACOSX
+static void PpidFromPid(pid_t procId, pid_t *parentProcId) {
+  proc_bsdinfo proc_info;
+  if (proc_pidinfo(procId, PROC_PIDTBSDINFO, 0, &proc_info, sizeof(proc_info)) > 0) {
+    *parentProcId = proc_info.pbi_ppid;
+  }
+}
+#endif
+
+static void PidFromPpid(pid_t parentProcId, pid_t **procId, int *size) {
+  std::vector<pid_t> vec; int i = 0;
+  #if CURRENT_PLATFORM_ID == OS_MACOSX
+  int cntp = proc_listpids(PROC_ALL_PIDS, 0, nullptr, 0);
+  std::vector<pid_t> proc_info(cntp);
+  std::fill(proc_info.begin(), proc_info.end(), 0);
+  proc_listpids(PROC_ALL_PIDS, 0, &proc_info[0], sizeof(pid_t) * cntp);
+  for (int j = cntp; j > 0; j--) {
+    if (proc_info[j] == 0) { continue; }
+    pid_t ppid; PpidFromPid(proc_info[j], &ppid);
+    if (ppid == parentProcId) {
+      vec.push_back(proc_info[j]); i++;
+    }
+  }
+  #elif CURRENT_PLATFORM_ID == OS_LINUX
+  PROCTAB *proc = openproc(PROC_FILLSTAT);
+  while (proc_t *proc_info = readproc(proc, nullptr)) {
+    if (proc_info->ppid == parentProcId) {
+      vec.push_back(proc_info->tgid); i++;
+    }
+    freeproc(proc_info);
+  }
+  closeproc(proc);
+  #elif CURRENT_PLATFORM_ID == OS_FREEBSD
+  int cntp; if (kinfo_proc *proc_info = kinfo_getallproc(&cntp)) {
+    for (int j = 0; j < cntp; j++) {
+      if (proc_info[j].ki_ppid == parentProcId) {
+        vec.push_back(proc_info[j].ki_pid); i++;
+      }
+    }
+    free(proc_info);
+  }
+  #endif
+  *procId = (pid_t *)malloc(sizeof(pid_t) * vec.size());
```

Comment:
Why are you copying the vector to a new buffer rather than just returning the vector?

Reply:
Sorry, this is based on code where i was already doing that and I forgot to make it C++ abi and not C.